### PR TITLE
Add an event for automatic updates settings changes in control center

### PR DESCRIPTION
**Pull request by pwithnall**
Signed-off-by: Philip Withnall <pwithnall@endlessos.org>

https://phabricator.endlessm.com/T29248

### CODE REVIEW

File: azafea/event_processors/endless/metrics/events/__init__.py [hunk 1]
```diff
@@ -76,6 +76,26 @@ class CacheMetadataIsCorrupt(SingularEvent):
...
 
+class ControlCenterAutomaticUpdates(SingularEvent):
+    __tablename__ = 'control_center_automatic_updates'
+    __event_uuid__ = '510f9741-823e-41a9-af2d-048895f990c0'
+    __payload_type__ = '(bbbv)'
+
+    allow_downloads_when_metered = Column(Boolean, nullable=False)
+    automatic_updates_enabled = Column(Boolean, nullable=False)
+    tariff_enabled = Column(Boolean, nullable=False)
+    tariff_variant = Column(Unicode, nullable=False)
```
Comment (wjt) [inline azafea/event_processors/endless/metrics/events/__init__.py:87]: Not json(b)?

Comment (wjt) [reply azafea/event_processors/endless/metrics/events/__init__.py:87]: (I believe the type is something like `(sua(suu))` with no types that should be problematic to convert to JSON?)

Comment (pwithnall) [reply azafea/event_processors/endless/metrics/events/__init__.py]: That’s a better solution, thanks. Even easier once I found [`GLib.Variant.unpack()`](https://lazka.github.io/pgi-docs/GLib-2.0/classes/Variant.html#GLib.Variant.unpack) to convert `GLib.Variant` to native Python types.

Comment (dbnicholson) [reply azafea/event_processors/endless/metrics/events/__init__.py]: Pygobject will also implicitly unpack it if you do various pythonic operations in it. Better to explicit though.

File: azafea/event_processors/endless/metrics/events/__init__.py [hunk 2]
```diff
@@ -76,6 +76,26 @@ class CacheMetadataIsCorrupt(SingularEvent):
...
+    tariff_enabled = Column(Boolean, nullable=False)
+    tariff_variant = Column(JSONB, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return {
+            'allow_downloads_when_metered': payload.get_child_value(0).get_boolean(),
+            'automatic_updates_enabled': payload.get_child_value(1).get_boolean(),
+            'tariff_enabled': payload.get_child_value(2).get_boolean(),
+            'tariff_variant': payload.get_child_value(3).unpack(),
```
Comment (wjt) [inline azafea/event_processors/endless/metrics/events/__init__.py:95]:
```
95: error: "Variant" has no attribute "unpack"
```

**Review by wjt — COMMENTED** [review]
I pushed a couple of patches which I think will fix the flake8 and mypy errors.

**Comment by wjt** [comment]
(Marking as draft because `fixup!` commits0

**Comment by pwithnall** [comment]
Those changes didn’t quite seem enough; I just pushed a couple more fixups which I think should get the linter passing.

**Review by wjt — APPROVED** [review]

**Comment by pwithnall** [comment]
Filed phabricator.endlessm.com/T31061 for deploying this to dev then prod.

**Pull request merged**